tests/influxdb/real_client_test.py

```diff
@@ -0,0 +1,560 @@
+
+        with open(conf_template) as fh:
+            conf = fh.read().format(
+                broker_port=self.influxd_broker_port,
+                admin_port=self.influxd_admin_port,
+                udp_port=self.influxd_udp_port,
+                broker_raft_dir=os.path.join(
+                    self.influxd_temp_dir_influxdb, 'raft'),
+                broker_node_dir=os.path.join(
+                    self.influxd_temp_dir_influxdb, 'db'),
+                influxdb_cluster_dir=os.path.join(
+                    self.influxd_temp_dir_influxdb, 'state'),
+                influxdb_logfile=self.logs_file
+            )
+
+        conf_file = os.path.join(self.influxd_temp_dir_base, 'influxdb.conf')
+        with open(conf_file, "w") as fh:
+            fh.write(conf)
+
+        # now start the server instance:
+        proc = self.influxdb_proc = subprocess.Popen(
+            [influxdb_bin_path, '-config', conf_file],
+            stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+
+        print("Started influxdb bin in %r with ports %s and %s.." % (
+              self.influxd_temp_dir_base,
+              self.influxd_admin_port, self.influxd_broker_port),
+              file=sys.stderr)
+
+        # wait for it to listen on the broker and admin ports:
+        # usually a fresh instance is ready in less than 1 sec ..
+        timeout = time.time() + 10  # so 10 secs should be rather enough,
+        # otherwise either your system load is rather high,
+        # or you run a 286 @ 1Mhz ?
+        try:
+            while time.time() < timeout:
+                if (is_port_open(self.influxd_broker_port)
+                        and is_port_open(self.influxd_admin_port)):
+                    break
+                time.sleep(0.5)
+                if proc.poll() is not None:
+                    raise RuntimeError('influxdb prematurely exited')
+            else:
+                proc.terminate()
+                proc.wait()
+                raise RuntimeError('Timeout waiting for influxdb to listen'
+                                   ' on its broker port')
+        except RuntimeError as err:
+            data = self.get_logs_and_output()
+            data['reason'] = str(err)
+            raise RuntimeError("%(reason)s. RC=%(rc)s\n"
+                               "stdout=%(out)r\nstderr=%(err)r\nlogs=%(logs)r"
+                               % data)
+
+    def get_logs_and_output(self):
+        proc = self.influxdb_proc
+        with open(self.logs_file) as fh:
+            return {
+                'rc': proc.returncode,
+                'out': proc.stdout.read(),
+                'err': proc.stderr.read(),
+                'logs': fh.read()
+            }
+
+    def close(self, remove_tree=True):
+        self.influxdb_proc.terminate()
+        self.influxdb_proc.wait()
+        if remove_tree:
+            shutil.rmtree(self.influxd_temp_dir_base)
+
+
+class TestClientWithNoServer(unittest.TestCase):
```

Comment:
effectively, had done a copy of client_test.py to start this one.. ;)
how about making `client_test.py` renamed to `client_tests_without_server.py`, and this one renamed to `client_tests_with_server.py` and also effectively remove this from here ?
or just rename this one to client_tests_with_server.py to insist enough ?